invenio_rdm_records/services/schemas/record.py

```diff
@@ -88,3 +90,37 @@ def default_nested(self, data, many, **kwargs):
             data["custom_fields"] = {}
 
         return data
+
+    def hide_tombstone_or_metadata(self, data):
+        """Hide tombstone info if the record isn't deleted and metadata if it is."""
+        if not data.get("deletion_status", {}).get("is_deleted", False):
+            data.pop("tombstone", None)
+        else:
+            data.pop("access", None)
+            data.pop("created", None)
+            data.pop("custom_fields", None)
+            data.pop("files", None)
+            data.pop("is_deleted", None)
+            data.pop("is_draft", None)
+            data.pop("is_published", None)
+            data.pop("media_files", None)
+            data.pop("metadata", None)
+            data.pop("parent", None)
+            data.pop("revision_id", None)
+            data.pop("stats", None)
+            data.pop("status", None)
+            data.pop("updated", None)
+            data.pop("versions", None)
```

Comment:
minor: would it make sense to summarize all these into something like "pop all fields that are not `tombstone`"? I could imagine that in the future we add a new field in the schema, and then forget to add it here (probably tests will catch this, but you never know :) )

Reply:
i've been toying with the idea too, but wasn't entirely sure about the set of fields to keep.
especially the `links` entries might be something worth keeping around`[1]`, so that the REST API stays machine-actionable.

the current set of keys that aren't thrown out here are:
`dict_keys(['id', 'links', 'pids', 'tombstone', 'deletion_status'])`

an example looks like:
```python
{'id': 'zm408-gyp73',
 'links': {'self': 'https://127.0.0.1/api/records/zm408-gyp73',
  'self_html': 'https://127.0.0.1/records/zm408-gyp73',
  'parent': 'https://127.0.0.1/api/records/806dg-1x022',
  'parent_html': 'https://127.0.0.1/records/806dg-1x022',
  'self_iiif_manifest': 'https://127.0.0.1/api/iiif/record:zm408-gyp73/manifest',
  'self_iiif_sequence': 'https://127.0.0.1/api/iiif/record:zm408-gyp73/sequence/default',
  'files': 'https://127.0.0.1/api/records/zm408-gyp73/files',
  'media_files': 'https://127.0.0.1/api/records/zm408-gyp73/media-files',
  'archive': 'https://127.0.0.1/api/records/zm408-gyp73/files-archive',
  'latest': 'https://127.0.0.1/api/records/zm408-gyp73/versions/latest',
  'latest_html': 'https://127.0.0.1/records/zm408-gyp73/latest',
  'draft': 'https://127.0.0.1/api/records/zm408-gyp73/draft',
  'versions': 'https://127.0.0.1/api/records/zm408-gyp73/versions',
  'access_links': 'https://127.0.0.1/api/records/zm408-gyp73/access/links',
  'reserve_doi': 'https://127.0.0.1/api/records/zm408-gyp73/draft/pids/doi',
  'communities': 'https://127.0.0.1/api/records/zm408-gyp73/communities',
  'communities-suggestions': 'https://127.0.0.1/api/records/zm408-gyp73/communities-suggestions',
  'requests': 'https://127.0.0.1/api/records/zm408-gyp73/requests'},
 'pids': {'oai': {'identifier': 'oai:my-site.com:zm408-gyp73',
   'provider': 'oai'}},
 'tombstone': {'note': '',
  'removed_by': None,
  'removal_date': '2023-08-04',
  'citation_text': '',
  'is_visible': True,
  'identifiers': [{'scheme': 'oai',
    'identifier': 'oai:my-site.com:zm408-gyp73'}]},
 'deletion_status': {'is_deleted': True, 'status': 'D'}}
```

are you okay with that set of fields?

`[1]` in a cut-down form of course, as e.g. the `files` links won't make much sense anymore given that they're not accessible anyway.
however, some links (like `self_html`) may still be useful to enable crawlers to navigate around.

Reply:
I think I'm ok with the list you shared so :+1: for me. It also resembles an explicit "allowed/safe" list of fields, vs. hand-removing fields that might contain information that we don't want to disclose.